src/test/java/org/elasticsearch/cluster/MinimumMasterNodesTests.java

```diff
@@ -126,29 +122,26 @@ public boolean apply(Object obj) {
         assertThat(state.nodes().size(), equalTo(2));
         assertThat(state.metaData().indices().containsKey("test"), equalTo(true));
 
-        logger.info("Running Cluster Health");
-        ClusterHealthResponse clusterHealth = client().admin().cluster().health(clusterHealthRequest().waitForGreenStatus()).actionGet();
-        logger.info("Done Cluster Health, status " + clusterHealth.getStatus());
-        assertThat(clusterHealth.isTimedOut(), equalTo(false));
-        assertThat(clusterHealth.getStatus(), equalTo(ClusterHealthStatus.GREEN));
+        ensureGreen();
 
-        logger.info("--> verify we the data back");
+        logger.info("--> verify we the data back after cluster reform");
         for (int i = 0; i < 10; i++) {
-            assertThat(client().prepareCount().setQuery(QueryBuilders.matchAllQuery()).execute().actionGet().getCount(), equalTo(100l));
+            assertHitCount(client().prepareCount().setQuery(QueryBuilders.matchAllQuery()).execute().actionGet(), 100);
         }
 
         internalCluster().stopRandomNonMasterNode();
-        assertThat(awaitBusy(new Predicate<Object>() {
+        assertBusy(new Runnable() {
             @Override
-            public boolean apply(Object obj) {
+            public void run() {
                 ClusterState state = client().admin().cluster().prepareState().setLocal(true).execute().actionGet().getState();
-                return state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID);
+                assertThat(state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID), equalTo(true));
             }
-        }), equalTo(true));
+        });
 
         logger.info("--> starting the previous master node again...");
         internalCluster().startNode(settings);
 
+        ensureGreen();
```

Comment:
Not sure I fully understand why are these ensureGreen were added. It seems that this ensureGreen is not needed since we are waiting for green in the next line and some other ensureGreen are either not needed (since we wait for number of active shards right after it) or they are making the tests less "evil" so to speak.